### PR TITLE
Save memory

### PR DESCRIPTION
**Pull request by FerreolS**
Do a few things to improve the memory footprint of optimization methods:
- previous iterate `xold` is no longer stored when it is not necessary 
- remove some temporary variable in ADMM

### CODE REVIEW

File: Opti/OptiADMM.m [hunk 1]
```diff
@@ -222,7 +218,8 @@ function initialize(this,x0)
                     this.xopt=this.CG.xopt;
                 end
             else
-                this.xopt=this.solver(this.zn,this.rho_n, this.xopt);
+                zn = this.yn{n}+this.wn{n};
```
Comment (esoubies) [inline Opti/OptiADMM.m:221]: I think that it should be zn = this.yn+this.wn;  (the solver requires all the n)

**Comment by esoubies** [comment]
Very good initiative ! ;)
Appart from the above comment, everything is fine on my side.
It may be of interest to run a code that uses the TestCvgADMM (I do not have one) for safety. @ThomasDeb ?

File: Opti/OptiADMM.m (hunk 1 above)
Comment (FerreolS) [reply Opti/OptiADMM.m:221]: Bien vu!
But there is no + for cells...

File: Example/DeconvEx/2D/Deconv_LS_TV_NonNeg.m [hunk 2]
```diff
@@ -64,7 +64,7 @@
 ADMM=OptiADMM(F,Fn,Hn,rho_n);
 ADMM.OutOp=OutputOptiSNR(1,im,10,[1 2]);
 % STOP when the sum successives C = F*x + Fn{1}*Hn{1}*x is lower than 1e-4 or when the distance between two successive step is lower than 1e-5
-ADMM.CvOp=TestCvgCombine(TestCvgCostRelative(1e-4,[1 2]), 'StepRelative',1e-4);  
+ADMM.CvOp=TestCvgCombine(TestCvgCostRelative(1e-4,[1 2]), 'StepRelative',1e-4,TestCvgADMM());
```
Comment (FerreolS) [inline Example/DeconvEx/2D/Deconv_LS_TV_NonNeg.m:67]: The test of `TestCvgADMM()` is working (even without `TestCvgCostRelative` and `StepRelative`

Comment (esoubies) [reply Example/DeconvEx/2D/Deconv_LS_TV_NonNeg.m:67]: :+1: Perfect ! Can merge :)
Thanks !

**Pull request merged**